geopandas/geoseries.py

```diff
@@ -53,6 +53,37 @@ def _geoseries_constructor_with_fallback(data=None, index=None, crs=None, **kwar
         return Series(data=data, index=index, **kwargs)
 
 
+def _geoseries_expanddim(data=None, index=None, crs=None, **kwargs):
+    from geopandas import GeoDataFrame
+
+    # DataFrame == pd.Series._constructor_expanddim
+    df = pd.DataFrame(data=data, index=index, **kwargs)
+    geo_col_name = None
+    if isinstance(data, GeoSeries):
+        # pandas default column name is 0, keep convention
+        geo_col_name = data.name if data.name is not None else 0
+
+    elif isinstance(data, dict) and len(data) == 1:
+        geo_col_name = list(data.keys())[0]
+    elif isinstance(data, BlockManager) and len(data.items) == 1:
+        geo_col_name = data.items[0]
```

Comment:
For which cases are those two checks needed? (would also be good to add some comments about it)

Reply:
Quickly checking, this is for `to_frame()`, right? (to ensure the series original name is set as geometry_column_name)

I am wondering if we can make this check less specific to what is exactly passed:

- We could generally check if `df.shape[1] == 1` (so the resulting DataFrame), and use its single column name. This would cover the `to_frame()` use case, I think. But would it be too generic? (would there be cases where it would set the geometry column that we don't want?)
- Specifically for `to_frame()`, we can probably also set the correct geometry name in `__finalize__` (if `other` is a GeoSeries)

Reply:
Dict is just `to_frame`, Block manager case I will need to go back throug the failed runs - I think this was for pandas main compatibility - I'd work out exactly, but this can be removed entirely on your suggestion below.

>We could generally check if df.shape[1] == 1 (so the resulting DataFrame), and use its single column name. This would cover the to_frame() use case, I think. But would it be too generic? (would there be cases where it would

Makes sense - I think these checks predate me having df = pd.DataFrame() at the top of the method. I can't see where this would fail, dimensions of `data` should be consistent with dimensions of `df` (except one is a dict/ blockmanager/ other) and other is dataframe.

>Specifically for to_frame(), we can probably also set the correct geometry name in `__finalize__` (if other is a GeoSeries)

As far as I can tell, to_frame doesn't call `__finalize__`, unless you mean that `_constructor_expanddim` should call it somehow?

Reply:
> As far as I can tell, to_frame doesn't call `__finalize__`, unless you mean that `_constructor_expanddim` should call it somehow?

At least on the main branch, this is actually being called:

https://github.com/pandas-dev/pandas/blob/2fe0c7065665f25da66a9a0a0a2a27c681812cb2/pandas/core/series.py#L1815-L1817

But it seems this was only added a few weeks ago (so for pandas 1.5). So that makes it a bit hard to rely on this of course.